nixops_aws/resources/s3_bucket.py

```diff
@@ -23,18 +44,16 @@ def get_resource_type(cls):
 
     def __init__(self, xml, config={}):
```

Comment:
```suggestion
    def __init__(self, name, config):
```

Having mutable defaults in arguments means that the same instance of the default is shared across every invocation of the class.
It's not a problem in this instance because it's always instantiated by NixOps always passes the correct things.
It's a common source of strange bugs in Python programs.

Reply:
Also the signature changed, name is `str`, and `config` is `nixops.resources.ResourceEval`.